server/gc/src/main/java/org/apache/accumulo/gc/GarbageCollectionAlgorithm.java

```diff
@@ -217,11 +221,50 @@ private void confirmDeletes(GarbageCollectionEnvironment gce,
         throw new RuntimeException(
             "Scanner over metadata table returned unexpected column : " + entry.getKey());
     }
+    Set<String> tableIdsAfter = gce.getTableIDs();
+    ensureAllTablesChecked(Collections.unmodifiableSet(tableIdsBefore),
+        Collections.unmodifiableSet(tableIdsSeen), Collections.unmodifiableSet(tableIdsAfter));
 
     confirmDeletesFromReplication(gce.getReplicationNeededIterator(),
         candidateMap.entrySet().iterator());
   }
 
+  @VisibleForTesting
+  /**
+   *
+   */
+  protected void ensureAllTablesChecked(Set<String> tableIdsBefore, Set<String> tableIdsSeen,
+      Set<String> tableIdsAfter) {
+
+    // if a table was added or deleted during this run, it is acceptable to not
+    // have seen those tables ids when scanning the metadata table. So get the intersection
+    Set<String> tableIdsMustHaveSeen = new HashSet<>(tableIdsBefore);
+    tableIdsMustHaveSeen.retainAll(tableIdsAfter);
+
+    if (tableIdsMustHaveSeen.isEmpty() && !tableIdsSeen.isEmpty()) {
```

Comment:
if tableIdsMustHaveSeen is empty, then that is enough to fail.  There should always be at least 1 table, i.e. the accumulo.metadata table.  It shouldn't matter if tableIdsSeen is empty or not.

Reply:
I thought there are separate runs for the Root, Metadata and User tables?

Reply:
Even with separate runs, this is still not possible. There will always be the replication table seen in the metadata table, and the metadata table will always be seen in the root table. `tableIdsMustHaveSeen` will always be non-empty, unless something seriously went wrong.